exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceTranslatorTest.java

```diff
@@ -0,0 +1,198 @@
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void nullTruncatableStringProtoTest(){
+    TruncatableString nullTruncatable = TraceTranslator.toTruncatableStringProto(null);
+
+    assertNull(nullTruncatable.getValue());
+    assertEquals(0, nullTruncatable.getTruncatedByteCount());
+  }
+
+  @Test
+  public void toTruncatableStringProtoTest(){
+    String truncatableString = "myTruncatableString";
+    TruncatableString testTruncatable = TraceTranslator.toTruncatableStringProto(truncatableString);
+
+    assertEquals("myTruncatableString", testTruncatable.getValue());
+    assertEquals(0, testTruncatable.getTruncatedByteCount());
+  }
+
+  @Test
+  public void toTimestampProtoTest(){
+    long epochNanos = TimeUnit.SECONDS.toNanos(3001) + 255;
+    com.google.protobuf.Timestamp timestamp = TraceTranslator.toTimestampProto(epochNanos);
+
+    assertEquals(3001, timestamp.getSeconds());
+    assertEquals(255, timestamp.getNanos());
+  }
+
+  @Test
+  public void toAttributesProtoTest(){
+    ReadableAttributes attributes = Attributes.newBuilder()
+            .setAttribute("myKey", "myValue")
+            .setAttribute("http.status_code", true)
+            .setAttribute("anotherKey", 100)
+            .setAttribute("http.host", 3.14)
+            .build();
+    Map<String, AttributeValue> fixedAttributes = new LinkedHashMap<>();
+    fixedAttributes.put("fixed", AttributeValue.newBuilder().setStringValue
+        (TruncatableString.newBuilder().setValue("attributes").setTruncatedByteCount(0).build()).build());
+    fixedAttributes.put("another", AttributeValue.newBuilder().setStringValue
+        (TruncatableString.newBuilder().setValue("entry").setTruncatedByteCount(0).build()).build());
+
+    Span.Attributes.Builder attributesBuilder = Span.Attributes.newBuilder().setDroppedAttributesCount(0)
+        .putAttributeMap("myKey", AttributeValue.newBuilder().setStringValue
+            (TruncatableString.newBuilder().setValue("myValue").setTruncatedByteCount(0)).build())
+        .putAttributeMap("/http/host", AttributeValue.newBuilder().setStringValue
+            (TruncatableString.newBuilder().setValue(String.valueOf(3.14)).setTruncatedByteCount(0)).build())
+        .putAttributeMap("/http/status_code", AttributeValue.newBuilder().setBoolValue
+            (true).build())
+        .putAttributeMap("anotherKey", AttributeValue.newBuilder().setIntValue
+            (100).build())
+        .putAttributeMap("g.co/agent", AttributeValue.newBuilder().setStringValue
+            (TruncatableString.newBuilder().setValue("opentelemetry-java [0.6.0]").setTruncatedByteCount(0)).build());
+    for (Map.Entry<String, AttributeValue> entry : fixedAttributes.entrySet()) {
+      attributesBuilder.putAttributeMap(entry.getKey(), entry.getValue());
+    }
+
+    Span.Attributes allAtributes = attributesBuilder.build();
+
+    assertEquals(allAtributes, TraceTranslator.toAttributesProto(attributes, fixedAttributes));
```

Comment:
I think the assertEquals will be sensitive to ordering changes and since ordering is not generally guaranteed for maps this may make this test hard to maintain or, worse, flaky. Some list matchers will allow you to specify that ordering does not matter, I don't recall if Junit4 has that option. Alternatively you can iterate over the list yourself. A slightly weaker test is preferable to a flaky one.

Reply:
That's a great point. I decided to iterate over the list manually. It's a bit lengthy, but it checks that every key that I expect to be in the attributeMap is actually in there, and that its respective value is what I expect it to be. I got rid of the original chunk of code leading up to the allAttributes variable.